--- /etc/xdg/luakit/binds.lua	2011-11-13 22:59:47.000000000 +0000
+++ binds.lua	2012-05-31 17:58:57.696339001 +0100
@@ -146,6 +146,16 @@
                                         if not m.count then w:scroll{ y = "0%" } else return false end
                                     end),
 
+    -- Toggle status bar visibility
+    key({},          "t",           function (w)
+        if true == w.sbar.hidden then
+            w.sbar.ebox:show() w.sbar.hidden = false
+        else
+            w.sbar.ebox:hide() w.sbar.hidden = true
+        end
+    end),
+
+
     -- Zooming
     key({},          "+",           function (w, m)    w:zoom_in(zoom_step  * m.count)       end, {count=1}),
     key({},          "-",           function (w, m)    w:zoom_out(zoom_step * m.count)       end, {count=1}),
@@ -171,9 +181,25 @@
                                         for i = 1, m.count do w:new_tab(w:search_open(uri)) end
                                     end, {count = 1}),
 
+    -- Play YouTube videos in mplayer using yturl
+    key({}, "v", function (w) 
+        local uri = w.view.hovered_uri or w.view.uri
+        if uri then
+            w:notify("Getting media URI for " .. uri)
+            local exit_status, stdout, stderr = luakit.spawn_sync("yturl " .. uri .. " 22 45 6 18")
+            w:notify("Video buffering...")
+            if exit_status == 0 then
+                luakit.spawn("mplayer -cache 8192 -cache-min 10 -- " .. stdout)
+                w:notify("Playing video.")
+            else
+                w:warning("yturl error: " .. stderr) 
+            end
+        end 
+    end),
+
     -- Yanking
     buf("^yy$",                     function (w)
-                                        local uri = string.gsub(w.view.uri or "", " ", "%%20")
+                                        local uri = string.gsub(w.view.hovered_uri or w.view.uri or "", " ", "%%20")
                                         luakit.selection.primary = uri
                                         luakit.selection.clipboard = uri
                                         w:notify("Yanked uri: " .. uri)
