--- /etc/xdg/luakit/globals.lua	2011-11-13 22:59:47.000000000 +0000
+++ globals.lua	2012-05-31 18:00:24.004339001 +0100
@@ -1,6 +1,6 @@
 -- Global variables for luakit
 globals = {
-    homepage            = "http://luakit.org/",
+    homepage            = "http://unix.stackexchange.com",
  -- homepage            = "http://github.com/mason-larobina/luakit",
     scroll_step         = 40,
     zoom_step           = 0.1,
@@ -68,12 +68,12 @@
 
 -- Per-domain webview properties
 -- See http://webkitgtk.org/reference/webkitgtk-WebKitWebSettings.html
-domain_props = { --[[
+domain_props = { 
     ["all"] = {
-        enable_scripts          = false,
-        enable_plugins          = false,
+        enable_scripts          = true,
+        enable_plugins          = true,
         enable_private_browsing = false,
-        user_stylesheet_uri     = "",
+        user_stylesheet_uri     = "file://" .. luakit.config_dir .. "/adblock.css",
     },
     ["youtube.com"] = {
         enable_scripts = true,
@@ -82,7 +82,7 @@
     ["bbs.archlinux.org"] = {
         user_stylesheet_uri     = "file://" .. luakit.data_dir .. "/styles/dark.css",
         enable_private_browsing = true,
-    }, ]]
+    },
 }
 
 -- vim: et:sw=4:ts=8:sts=4:tw=80
